src/main/java/com/aws/greengrass/security/SecurityService.java

```diff
@@ -163,6 +166,46 @@ public KeyManager[] getKeyManagers(URI privateKeyUri, URI certificateUri)
         return provider.getKeyManagers(privateKeyUri, certificateUri);
     }
 
+    /**
+     * Get certificate chain using private and certificate URIs.
+     *
+     * @param privateKeyUri  private key URI
+     * @param certificateUri certificate URI
+     * @return X509Certificate list that corresponds to the given private key and certificate URIs
+     * @throws ServiceUnavailableException      if crypto key provider service is unavailable
+     * @throws KeyLoadingException              if crypto key provider service fails to load key
+     * @throws CertificateChainLoadingException if no certificate chain is found
+     */
+    public X509Certificate[] getCertificateChain(URI privateKeyUri, URI certificateUri)
+            throws CertificateChainLoadingException, ServiceUnavailableException, KeyLoadingException {
+        logger.atTrace().kv(KEY_URI, privateKeyUri).kv(CERT_URI, certificateUri)
+                .log("Getting the certificate chain using private key and certificate URIs");
+        CryptoKeySpi provider = selectCryptoKeyProvider(privateKeyUri);
+        KeyManager[] km = provider.getKeyManagers(privateKeyUri, certificateUri);
+
+        if (km.length == 1 && km[0] instanceof X509KeyManager) {
+            X509KeyManager x509KeyManager = (X509KeyManager) km[0];
+            KeyPair keyPair = provider.getKeyPair(privateKeyUri, certificateUri);
+            String[] aliases = x509KeyManager.getClientAliases(keyPair.getPublic().getAlgorithm(), null);
+            if (aliases == null) {
+                logger.atError().kv(KEY_URI, privateKeyUri).kv(CERT_URI, certificateUri)
+                        .log("Unable to find aliases in the key manager.");
+                throw new CertificateChainLoadingException("Unable to get the certificate chain from the provider "
+                        + "using the key manager");
+            }
+            for (String alias : aliases) {
+                if (x509KeyManager.getPrivateKey(alias).equals(keyPair.getPrivate())) {
+                    return x509KeyManager.getCertificateChain(alias);
+                }
+            }
+        }
+        logger.atError().kv(KEY_URI, privateKeyUri).kv(CERT_URI, certificateUri)
+                .log("Unable to get the certificate chain using private key and certificate URIs");
+        throw new CertificateChainLoadingException("Unable to get the certificate chain from the provider using the "
+                + "key manager");
```

Comment:
Only throw. And have multiple exception messages based on what specifically went wrong. We need to be able to diagnose this without access to the customer's device. 

In this place, this exception can happen for more than once reason which will be tough to figure out.